src/Agoda.IoC.AutofacExt/AutoWireAssemblyExt.cs

```diff
@@ -87,11 +87,23 @@ public static class AutoWireAssemblyExt
                             if (toType.IsGenericTypeDefinition)
                             {
                                 services.RegisterGeneric(toType).As(reg.FromType).SingleInstance();
-
                             }
                             else
                             {
-                                services.RegisterType(toType).As(reg.FromType).SingleInstance();
+                                
+                                if (toType.GetInterfaces().Any(x => x == typeof(IStartable)))
```

Comment:
As far as I understand Autofac does not impose a restriction that IStartable classes must be singletons (only advises against having any other lifetime). Is it intentional that we only register singletons?

Reply:
Yep, this is a stop gap to help a project migrate to netcore, i think long term we wont support autofac, as it has most of these feature we are trying to build

Reply:
so yes you are correct that autofac doesnt allow only singletons to support startable